contracts/mocks/libraries/Validation.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import '../../../contracts/hts-precompile/HederaResponseCodes.sol';
+
+library Validation {
```

Comment:
Hey @Nana-EC do you think this would be a good way to refactor all validation in the `HtsPrecompileMock`?

So define all validation functions in this library and have the HederaResponseCodes returned here, and if this is taken to its fullest extent `HederaResponseCodes` will never be used directly in `HtsPrecompileMock.sol`

Though this approach may make [task 2](https://github.com/hashgraph/hedera-json-rpc-relay/issues/1220) difficult if hedera-services doesn't neatly follow a validation process similar to that being done here.

> Investigate the ordering of response codes on Hedera and adjust the ordering in the mocks accordingly.

Reply:
Hmm, yeah I like the idea but if you're attempting to mimic the behaviour of hedera-services internally then we face diverging logic if as you said something changes.

Interestingly enough what you have is very unlikely to change it just may not be the final response code emitted. If a separate issue is caught before checking for a deleted token this would fail.

I would leave any assumption of response code to the the test level or made the function specific to the item e.g. validateDeleteTokenResponse

Reply:
@Nana-EC ok, so do you think I proceed with this refactoring for the rest of the HederaResponseCodes in HtsPrecompileMock?

Reply:
@Nana-EC guess you aren't active on weekends? so I went ahead and did the validation refactoring. I don't think I'm going to make any more changes to the mocks for now(might still improve the tests though).

If you're fine with these changes, I'll merge this PR.

Reply:
Hey @mshakeg , i think the changes are good so you can merge this PR